experiment/commenter/main.go

```diff
@@ -0,0 +1,224 @@
+	re := regexp.MustCompile(`.+/(.+)/(.+)/(issues|pull)/(\d+)$`)
+	mat := re.FindStringSubmatch(url)
+	if mat == nil {
+		return "", "", 0, fmt.Errorf("failed to parse: %s", url)
+	}
+	n, err := strconv.Atoi(mat[4])
+	if err != nil {
+		return "", "", 0, err
+	}
+	return mat[1], mat[2], n, nil
+}
+
+func makeQuery(query string, includeClosed bool, minUpdated time.Duration) (string, error) {
+	parts := []string{query}
+	if !includeClosed {
+		if strings.Contains(query, "is:closed") {
+			return "", fmt.Errorf("--query='%s' containing is:closed requires --include-closed", query)
+		}
+		parts = append(parts, "is:open")
+	} else if strings.Contains(query, "is:open") {
+		return "", fmt.Errorf("--query='%s' should not contain is:open when using --include-closed", query)
+	}
+	if minUpdated != 0 {
+		latest := time.Now().Add(-minUpdated)
+		parts = append(parts, "updated:<="+latest.Format(time.RFC3339))
+	}
+	return strings.Join(parts, " "), nil
+}
+
+type client interface {
+	CreateComment(owner, repo string, number int, comment string) error
+	FindIssues(query, sort string, asc bool) ([]github.Issue, error)
+}
+
+func main() {
+	log.SetFlags(log.LstdFlags | log.Lshortfile)
+	o := flagOptions()
+
+	if o.query == "" {
+		log.Fatal("empty --query")
+	}
+	if o.token == "" {
+		log.Fatal("empty --token")
+	}
+	if o.comment == "" {
+		log.Fatal("empty --comment")
+	}
+	b, err := ioutil.ReadFile(o.token)
+	if err != nil {
+		log.Fatalf("cannot read --token: %v", err)
+	}
+
+	var c client
+	tok := strings.TrimSpace(string(b))
+	if o.confirm {
+		c = github.NewClient(ClientUser, tok)
+	} else {
+		c = github.NewDryRunClient(ClientUser, tok)
+	}
+
+	query, err := makeQuery(o.query, o.includeClosed, o.updated)
+	if err != nil {
+		log.Fatalf("Bad query: %v", err)
+	}
+	sort := ""
+	asc := false
+	if o.updated > 0 {
+		sort = "updated"
+		asc = true
+	}
+	commenter := makeCommenter(o.comment, o.useTemplate)
+	if err := run(c, query, sort, asc, commenter, o.ceiling); err != nil {
+		log.Fatalf("Failed run: %v", err)
+	}
+}
+
+func makeCommenter(comment string, useTemplate bool) func(meta) (string, error) {
+	if !useTemplate {
+		return func(_ meta) (string, error) {
+			return comment, nil
+		}
+	}
+	t := template.Must(template.New("comment").Parse(comment))
+	return func(m meta) (string, error) {
+		out := bytes.Buffer{}
+		err := t.Execute(&out, m)
+		return string(out.Bytes()), err
+	}
+}
+
+func run(c client, query, sort string, asc bool, commenter func(meta) (string, error), ceiling int) error {
+	log.Printf("Searching: %s", query)
+	issues, err := c.FindIssues(query, sort, asc)
+	if err != nil {
+		return fmt.Errorf("search failed: %v", err)
+	}
+	problems := []string{}
+	log.Printf("Found %d matches", len(issues))
+	for n, i := range issues {
+		if ceiling > 0 && n == ceiling {
+			log.Printf("Stopping at --ceiling=%d of %d results", n, len(issues))
+			break
+		}
+		log.Printf("Matched %s: %s", i.HTMLURL, i.Title)
```

Comment:
Add a space between the url and the colon or switch %s to %q, otherwise the link that's created, doesn't get you in the PR.